tests/tests_change_settings.yml

```diff
@@ -0,0 +1,77 @@
+- hosts: all
+  tasks:
+    - name: Ensure required packages are installed
+      package:
+        name: tuned
+        state: present
+
+    - name: Ensure kernel settings profile directory exists
+      file:
+        path: /etc/tuned/kernel_settings
+        state: directory
+        mode: 0755
+
+    - name: Generate a configuration for kernel settings
+      copy:
+        src: tuned/etc/tuned/basic_settings/tuned.conf
+        dest: /etc/tuned/kernel_settings/tuned.conf
+        mode: 0644
+
+    - name: Ensure required services are enabled and started
+      service:
+        name: tuned
+        state: restarted
+        enabled: yes
+
+    - name: apply kernel_settings
+      include_role:
+        name: linux-system-roles.kernel_settings
+      vars:
+        kernel_settings_sysctl:
+          - name: fs.file-max
+            value: 400000
+          - name: kernel.threads-max
+            state: absent
+        kernel_settings_sysfs:
+          - name: /sys/kernel/debug/x86/nmi_longest_ns
+            value: 2000000
+        kernel_settings_bootloader_cmdline:
+          - name: spectre_v2
+            value: "on"
+          - name: quiet
+            state: absent
+
+    - name: reboot the machine - see if settings persist after reboot
+      reboot:
+        test_command: tuned-adm active
+
+    - name: check after reboot for ns_longest
+      shell: grep 2000000 /sys/kernel/debug/x86/nmi_longest_ns
+      
+    - name: check after reboot cmdline
+      shell: grep spectre_v2 /proc/cmdline
```

Comment:
I think `tuned-adm verify -i` checks this too.

Reply:
Should we rely on tuned backend?

Reply:
> Should we rely on tuned backend?

For now, I would say yes, until such time as we support a different provider (we don't call them backends, we call them providers).  However, if you feel that we should explicitly call `sysctl` to check `sysctl` parameters, look at files under `/sys` to check `sysfs` parameters, etc. then that's fine.

I don't think we have general guidelines for provider specific testing - @pcahyna please correct me if I'm wrong.  For example - the network role has two providers - NetworkManager (`nm`) and SysV Init scripts (`initscripts`) - https://github.com/linux-system-roles/network/tree/master/tests - As you can see here there are tests like `tests_bridge_nm.yml` and `tests_bridge_initscripts.yml` - these set the `network_provider: nm` or `network_provider: initscripts` then run the role and verify the results.  The code that does the verification seems mostly to be written to handle both providers.  For example - https://github.com/linux-system-roles/network/blob/master/tests/tasks/get_profile_stat.yml - handles both the `nm` case and the `initscripts` case.

Reply:
Ok. Added to test.
So kernel roles are now tuned testing.